omrsigcompat/omrsig.cpp

```diff
@@ -333,7 +333,11 @@ omrsig_signalOS_internal(int signum, const struct sigaction *act, struct sigacti
 	if (NULL == signalOS) {
 		rc = -1;
 	} else {
-		sighandler_t old = signalOS(signum, act->sa_handler);
+		sighandler_t handler = NULL;
+		if (NULL != act) {
+			handler = act->sa_handler;
+		}
+		sighandler_t old = signalOS(signum, handler);
```

Comment:
Have you tested this on a Windows machine?  What was the result?

My reading of the [msdn doc](https://msdn.microsoft.com/en-us/library/xdkz3x12.aspx) for `signal` indicates it can only be passed `SIG_IGN`, `SIG_DFL`, or a user function.

Unless `SIG_IGN` or `SIG_DFL` map to NULL, based on the doc I don't think this is valid and expect it to return `SIG_ERR`.

One ugly workaround would be to set it to SIG_IGN, get the old handler, and then set the old handler back.  I don't like that kind of hack though it opens a window where the "wrong" signal handler is installed.

Reply:
Tested on Windows:

Passing NULL to handler, signal() does not return SIG_ERR, it returns a valid value. 
SIG_DFL is 0, SIG_IGN is 1.

Reply:
Probably we shouldn't let user query the handler by passing NULL act. As SIG_DFL is mapped to NULL. Query using NULL act would accidentally set signal action to SIG_DFL